### ISSUE TITLE
Flash not working on Redmi Note 8 Pro

### ISSUE THREAD

**Issue by Gl0deanR** [yuriy-budiyev/code-scanner#153]
I have made a demo app for my school project and it works just fine until I tested it on a Xiaomi Redmi Note 8 Pro running MUI Global v12.0.5 Stable, Android 10.
Tested the app on several Samsung phones and everything works.
Permissions and all, focus, flashlight, scanning.
Except when I installed it on a  Xiaomi Redmi Note 8 Pro. Here, everything but the flashlight work perfectly.

The only errors I see in Android Studio when running on debug mode are:

E/GraphicExt: GraphicExtModuleLoader::CreateGraphicExtInstance false

E/libc: Access denied finding property "camera.aux.packagelist"

E/libc: Access denied finding property "vendor.camera.hal1.packagelist"

I am using the latest version of the code scanner:
`implementation("com.github.yuriy-budiyev:code-scanner:2.1.2")`

And I am using the permissions:
`<uses-permission android:name="android.permission.CAMERA">`
and I have also tried with:
`<uses-feature android:name="android.hardware.camera">`

I would appreciate so, so much if anyone found or finds a fix to this.

Thank you!

**Comment by HunterPillu** [comment]
[Gl0deanR](https://github.com/Gl0deanR) did you find any solution??. I am getting same error on Samsung S9 Android 10.

**Comment by Gl0deanR** [comment]
> [Gl0deanR](https://github.com/Gl0deanR) did you find any solution??. I am getting same error on Samsung S9 Android 10.

No... I'm still waiting for @yuriy-budiyev response, maybe he will know ho to fix this for us.

**Comment by mahendradabiaplite** [comment]
I'm also having the same issue with some device Vivo x70 pro

**Comment by HunterPillu** [comment]
@mahendradabiaplite 
could you please try again by adding 
<uses-feature android:name="android.hardware.camera.any">
 and 
<uses-feature android:name="android.hardware.camera.flashlight">

**Comment by mahendradabiaplite** [comment]
@HunterPillu  not working after adding this permission too in vivo x70 pro

**Comment by mpierucci** [comment]
@yuriy-budiyev  any update on this?

**Comment by RuiEEE** [comment]
Also not working on Samsung A22 5G

**Comment by Wolfizarbabalam** [comment]
Also not working on Sony Xperia XZ3 and Samsung A03s

**Comment by janakbist** [comment]
val scannerView = findViewById<CodeScannerView>(R.id.scanner_view)
codeScanner = CodeScanner(this, scannerView)
codeScanner.isFlashEnabled = true

.....................................................is there another way to enable the flash light

**Comment by DarkCode462** [comment]
an also not working on Redmi 10 s android 12 - S

**Comment by slmg-it-dept** [comment]
Also not working in Samsung A04e

**Comment by bahadirerdemm** [comment]
did you find a solution

**Comment by slmg-it-dept** [comment]
> did you find a solution


    
you can use this instead of that
<com.journeyapps.barcodescanner.CompoundBarcodeView
        android:id="@+id/barcode_scanner"
        android:layout_width="match_parent"
        android:layout_height="match_parent"/>

**Comment by slmg-it-dept** [comment]
> com.journeyapps.barcodescanner.CompoundBarcodeView

implementation 'com.journeyapps:zxing-android-embedded:4.3.0'
